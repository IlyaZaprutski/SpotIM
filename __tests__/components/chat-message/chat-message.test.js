import React from 'react';
import { MuiMountWithContext } from '../../test-utils';

import ChatMessage from '../../../src/js/components/chat-message/chat-message';

describe('ChatMessage component', () => {
    let chatMessage = null;

    const baseProps = {
        userName: 'ilya',
        avatarUrl: 'https://spotim-demo-chat-server.herokuapp.com/avatars/001-snorlax.png',
        messageText: 'qwertyui',
        isOwnMessage: true,
        date: 0,
    };

    it('mounts without crashing', () => {
        chatMessage = <ChatMessage {...baseProps} />;

        const chatMessageComponent = MuiMountWithContext(chatMessage);
        expect(chatMessageComponent).toMatchSnapshot();
    });

    it('should render a user name and messageText', () => {
        chatMessage = <ChatMessage {...baseProps} />;

        const chatMessageComponent = MuiMountWithContext(chatMessage);

        expect(chatMessageComponent.prop('userName')).toEqual('ilya');
        expect(chatMessageComponent.prop('messageText')).toEqual('qwertyui');
    });

    it('check class names', () => {
        chatMessage = <ChatMessage {...baseProps} classNames="test-class" isOptimistic />;

        const chatMessageComponent = MuiMountWithContext(chatMessage);
        const chatMessageNode = chatMessageComponent.find('.chat-message');

        expect(chatMessageNode.hasClass('test-class')).toEqual(true);
        expect(chatMessageNode.hasClass('chat-message_own-message')).toEqual(true);
        expect(chatMessageNode.hasClass('chat-message_optimistic')).toEqual(true);
    });
});
