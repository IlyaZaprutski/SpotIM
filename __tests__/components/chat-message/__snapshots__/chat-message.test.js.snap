// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatMessage component mounts without crashing 1`] = `
<ChatMessage
  avatarUrl="https://spotim-demo-chat-server.herokuapp.com/avatars/001-snorlax.png"
  classNames=""
  date={0}
  isOptimistic={false}
  isOwnMessage={true}
  messageText="qwertyui"
  userName="ilya"
>
  <div
    className="chat-message chat-message_own-message"
  >
    <div
      className="chat-message__avatar-container"
    >
      <Avatar
        className="chat-message__avatar"
        size={50}
        src="https://spotim-demo-chat-server.herokuapp.com/avatars/001-snorlax.png"
      >
        <img
          className="chat-message__avatar"
          size={50}
          src="https://spotim-demo-chat-server.herokuapp.com/avatars/001-snorlax.png"
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "rgb(188, 188, 188)",
              "borderRadius": "50%",
              "color": "#ffffff",
              "display": "inline-flex",
              "fontSize": 25,
              "height": 50,
              "justifyContent": "center",
              "muiPrepared": true,
              "userSelect": "none",
              "width": 50,
            }
          }
        />
      </Avatar>
    </div>
    <div
      className="chat-message__message-info"
    >
      <div
        className="chat-message__header"
      >
        <span
          className="chat-message__user-name"
        >
          ilya
        </span>
        <span
          className="chat-message__date"
        >
          <TimeAgo
            classNames=""
            date={0}
            isLiveUpdate={true}
          >
            <time
              className="time-ago"
            >
              48 years ago
            </time>
          </TimeAgo>
        </span>
      </div>
      <div
        className="chat-message__text"
      >
        qwertyui
      </div>
    </div>
  </div>
</ChatMessage>
`;
